coderd/database/migrations/000151_workspace_agent_auth_tokens_unique_idx.up.sql

```diff
@@ -0,0 +1,25 @@
+-- Firstly, we need to drop the old index.
+DROP INDEX IF EXISTS workspace_agents_auth_token_idx;
+
+-- Secondly, we need to fix any duplicate auth_tokens.
+-- We do this by setting the auth_token of the duplicate agent to a new UUID.
+UPDATE
+	workspace_agents
+SET
+	auth_token = gen_random_uuid()
+WHERE
+	id
+IN (
+	SELECT wa1.id
+	FROM
+		workspace_agents wa1
+	INNER JOIN
+		workspace_agents wa2
+	ON
+		wa1.id != wa2.id
```

Comment:
Suggestion: Let's stabilize the join, otherwise both will be updated, still an edge case with both created at exactly the same time but...
```suggestion
		wa1.id != wa2.id
		AND wa1.created_at <= wa2.created_at
```